login_test.py

```diff
@@ -56,6 +54,13 @@ def get_php_password_hash(plaintext_password):
             )
     return result.stdout.decode("UTF-8")
 
+def wait_for_redirect(driver):
+    previous_url = driver.current_url
+    def url_has_changed(driver):
+        return driver.current_url != previous_url
+    WebDriverWait(driver, 10).until(url_has_changed)

```

Comment:
Is there a reason that you didn't use the built-in `url_changes`?

Reply:
I think I read "exception" instead of "expectation" in the documentation and than thought it was some kind of error-handling. Will fix that.